### DISCUSSION TITLE
Search for the same embedding, return different distances

### DISCUSSION THREAD

**Discussion by rky0930** [milvus-io/milvus#4935] [Q&A and General discussion] [answered]
Search for the same embedding, return different distances
The same embeddings exist with different ids in my milvus database, but when I search with the embeddings, each distance is returned differently. I use IVF_PQ and there are multiple segments in one partition.

The reason why different distances are returned for the same embedding is because the embeddings exist in different segments, and the codebook used for product quantization in each segment is different?

Is there any way to always return the same distance for the same embedding when searching in milvus?

**Comment by shengjun1985** [comment, marked as answer]
> The reason why different distances are returned for the same embedding is because the embeddings exist in different segments, and the codebook used for product quantization in each segment is different?

Yes.

> Is there any way to always return the same distance for the same embedding when searching in milvus?

At the moment, there may be no way for IVF_PQ or IVF_SQ in multi segments.
Can you put the whole data in one segment?
Or do you accept IVF_FLAT?

**Reply by rky0930** [reply]
Thanks for answering !

> Can you put the whole data in one segment?
> Or do you accept IVF_FLAT?

Since the data to be stored exceeds 4GB, it seems difficult to put it in one segment.
IVF_FLAT seems to be impossible to use due to memory.
